code/game/objects/items/explosives/mine.dm

```diff
@@ -168,38 +227,61 @@
 
 
 /obj/item/explosive/mine/proc/set_tripwire()
-	if(!active && !tripwire)
+	if(!active && !tripwire && use_tripwire)
 		var/tripwire_loc = get_turf(get_step(loc, dir))
 		tripwire = new(tripwire_loc)
 		tripwire.linked_claymore = src
-		active = TRUE
+	active = TRUE
+
 
 
 //Mine can also be triggered if you "cross right in front of it" (same tile)
-/obj/item/explosive/mine/Crossed(atom/A)
+/obj/item/explosive/mine/Crossed(atom/victim)
 	..()
-	if(isliving(A))
-		var/mob/living/L = A
-		if(!L.stat == DEAD)//so dragged corpses don't trigger mines.
-			return
-		else
-			try_to_prime(A)
+	try_to_prime(victim)
 
-/obj/item/explosive/mine/Collided(atom/movable/AM)
-	try_to_prime(AM)
+/obj/item/explosive/mine/Collided(atom/movable/atom_movable)
+	try_to_prime(atom_movable)
 
+/**
+ * Proc that notifies a heavy trigger failing to detonate
+ */
+/obj/item/explosive/mine/proc/heavy_trigger_notify(mob/living/unfortunate_soul)
+	unfortunate_soul.visible_message(SPAN_DANGER("[icon2html(src, viewers(src))] [name] clicks as [unfortunate_soul] moves on top of it."), \
+	SPAN_DANGER("[icon2html(src, unfortunate_soul)] [name] clicks as you move on top of it."), \
+	SPAN_DANGER("You hear a click."))
 
-/obj/item/explosive/mine/proc/try_to_prime(mob/living/L)
+/**
+ * Proc that runs checks before priming occurs
+ */
+/obj/item/explosive/mine/proc/try_to_prime(mob/living/unfortunate_soul)
 	if(!active || triggered || (customizable && !detonator))
 		return
-	if(!istype(L))
+	if(!istype(unfortunate_soul))
 		return
-	if(L.stat == DEAD)
+	if(unfortunate_soul.stat == DEAD)
 		return
-	if(L.get_target_lock(iff_signal) || isrobot(L))
+	if(unfortunate_soul.get_target_lock(iff_signal) || isrobot(unfortunate_soul))
 		return
-	L.visible_message(SPAN_DANGER("[icon2html(src, viewers(src))] The [name] clicks as [L] moves in front of it."), \
-	SPAN_DANGER("[icon2html(src, L)] The [name] clicks as you move in front of it."), \
+	if(heavy_trigger && buried)
+		playsound(loc, 'sound/weapons/flipblade.ogg', 35, 1)
+		if(isxeno(unfortunate_soul))
+			var/mob/living/carbon/xenomorph/xeno = unfortunate_soul
+			if(xeno.mob_size < MOB_SIZE_BIG)
+				heavy_trigger_notify(unfortunate_soul)
+				return
+		if(prob(75) && ishuman(unfortunate_soul))
+			var/mob/living/carbon/human/human = unfortunate_soul
+			if(!human.wear_suit)
+				heavy_trigger_notify(unfortunate_soul)
+				return
+			if(human.wear_suit.slowdown > SLOWDOWN_ARMOR_MEDIUM) // "Nice hustle, 'tons-a-fun'! Next time, eat a salad!"
+				heavy_trigger_notify(unfortunate_soul)
+				return
+
+
+	unfortunate_soul.visible_message(SPAN_DANGER("[icon2html(src, viewers(src))] [name] clicks as [unfortunate_soul] moves on top of it."), \
+	SPAN_DANGER("[icon2html(src, unfortunate_soul)] [name] clicks as you move on top of it."), \
 	SPAN_DANGER("You hear a click."))
```

Comment:
Looks like a leftover? This seems like it should already be done by `heavy_trigger_notify`